tests/Fixer/LanguageConstruct/SingleSpaceAfterConstructFixerTest.php

```diff
@@ -0,0 +1,2617 @@
+                'trait',
+            ],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixWithTraitCases()
+    {
+        return [
+            [
+                '<?php trait Foo {}',
+                '<?php trait  Foo {}',
+            ],
+            [
+                '<?php trait Foo {}',
+                '<?php trait
+
+Foo {}',
+            ],
+            [
+                '<?php trait /* foo */Foo {}',
+                '<?php trait  /* foo */Foo {}',
+            ],
+            [
+                '<?php trait /* foo */Foo {}',
+                '<?php trait/* foo */Foo {}',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideFixWithTryCases
+     *
+     * @param string      $expected
+     * @param null|string $input
+     */
+    public function testFixWithTry($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'constructs' => [
+                'try',
+            ],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixWithTryCases()
+    {
+        return [
+            [
+                '<?php try {} catch (\Exception $exception) {}',
+                '<?php try{} catch (\Exception $exception) {}',
+            ],
+            [
+                '<?php try {} catch (\Exception $exception) {}',
+                '<?php try  {} catch (\Exception $exception) {}',
+            ],
+            [
+                '<?php try {} catch (\Exception $exception) {}',
+                '<?php try
+
+{} catch (\Exception $exception) {}',
+            ],
+            [
+                '<?php try /* foo */{} catch (\Exception $exception) {}',
+                '<?php try/* foo */{} catch (\Exception $exception) {}',
+            ],
+            [
+                '<?php try /* foo */{} catch (\Exception $exception) {}',
+                '<?php try  /* foo */{} catch (\Exception $exception) {}',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideFixWithUseCases
+     *
+     * @param string      $expected
+     * @param null|string $input
+     */
+    public function testFixWithUse($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'constructs' => [
+                'use',
+            ],
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixWithUseCases()
+    {
+        return [
+            [
+                '<?php use Foo\Bar;',
+                '<?php use  Foo\Bar;',
+            ],
+            [
+                '<?php use Foo\Bar;',
+                '<?php use
+
+Foo\Bar;',
+            ],
+            [
+                '<?php use /* foo */Foo\Bar;',
+                '<?php use/* foo */Foo\Bar;',
+            ],
+            [
+                '<?php use /* foo */Foo\Bar;',
+                '<?php use  /* foo */Foo\Bar;',
+            ],
+            // not detected as T_USE, why?
```

Comment:
@julienfalque @keradus SpacePossum 

Does anyone have an idea why these cases are not detected?

Unless I'm mistaken, these usages of `use` should also be `T_USE`, shouldn't they?

Reply:
This is because `PhpCsFixer\Tokenizer\Transformer\UseTransformer` changes those to `CT::T_USE_TRAIT` and `CT::T_USE_LAMBDA`.

Reply:
@julienfalque 

Ha, good to know!

I will add these as separate tokens, then!